### ISSUE TITLE
[client-side-base-visitor]   Allow importing documents from an external source - making it possible to use webpack loaders

### ISSUE THREAD

**Issue by eddeee888** [dotansimha/graphql-code-generator#2210]
**Is your feature request related to a problem? Please describe.**

Allow importing document nodes from external such as`.graphql`, `.ts` files. This makes it more flexible to import documents. If this is done, we can utilise the [webpack loader](https://www.apollographql.com/docs/react/recipes/webpack/)!

**Describe the solution you'd like**

Given the following document:

```graphql
mutation Login($input: LoginInput!) {
  login(input: $input) {
    id
    email
  }
}
```

We should be able to generate the following:

```tsx
// Login.generated.tsx
import * as Types from "../__generated__/types";
import * as operations from "./Login.graphql"; // <-- This is the line we are generating

// Other imports / types we already generate

export const LoginComponent = (props: LoginComponentProps) => {
  <ReactApollo.Mutation<LoginMutation, LoginMutationVariables> 
    mutation={operations.Login} {...props} />
}
```

Most of these types are already generated by the existing plugins/presets (Which is awesome!). 

The only thing blocking this is [here](https://github.com/dotansimha/graphql-code-generator/blob/master/packages/plugins/other/visitor-plugin-common/src/client-side-base-visitor.ts#L253). This means we always parse the document string generated by stringifying the parsed document or use the `graphql-tag` to generate the document at run time.

My proposal is that we add a config option (naming could be changed): `documentSource` which could be `graphQLTag`, `documentNode` or `external`:

- `graphQLTag`: is the default option i.e. importing `graphql-tag` and parsing document at runtime using `gql`
- `documentNode`: is the equivalent of `noGraphQLTag`
- `external`: import documents from a file. This could be a `.graphql` file for the case of webpack. Documents could also come from a `.ts` or `.tsx` file. e.g. users may want to export parsed document manually. When we use setting, we simply don't include the `documentString` in the generated template.

**Usage for `external`**

`external` mode could have the following API:

- `.graphql#near-operation-file`: This means we get the documents from the graphql file next to the file. For this mode, we will have to pass the documents from params into the plugin [here](https://github.com/dotansimha/graphql-code-generator/blob/f074de1b64cb489dc02ba199f73d9b2e066ac4a1/packages/plugins/typescript/react-apollo/src/index.ts#L182)

- `path/to/file/ABC.ts`, `path/to/file/ABC.graphql`: This means we get all documents from a certain file.

**Problems**

This means there's going to be a change in API and I don't know where `client-side-base-visitor.ts` is used so it would need to be carried onto whichever plugin extending it.

**Describe alternatives you've considered**

We could create just another setting: `importParsedDocumentsFrom` which behaves the same way. However, by doing this, we don't know that this could be related to `noGraphQLTag` and there's a chance someone will use them together. 

**Additional context**

I have got most of it working locally, happy to open a PR

**Comment by dotansimha** [comment]
Hi @eddeee888 ! That's sounds like a good idea (we always welcome PRs 😉 )

**Comment by dotansimha** [comment]
Available in 1.5.0 🎉
